scripts/camera.lua

```diff
@@ -292,4 +310,59 @@ function Camera.setTransitionPeriod(camera, transitionPeriod)
     Camera.updateConfig(camera)
 end
 
+--- @param player LuaPlayer
+--- @param camera Camera.camera
+function Camera.updateChartTags(player, camera)
+    -- we can't do this without a player or force
+    if not player or not player.force then
+        return
+    end
+    
+    -- we have duplicated calculations for recovering tile coordinates from center/zoom
+    -- if we had a third box, I think it would be worth extracting this logic/data somehow
```

Comment:
I think it would already pay off to do in this PR, as you can have multiple calls to this factored code from the places where the camera position got updated (e.g. [here](https://github.com/veger/TLBE/blob/e34d5ee52662a4eeda0c95366f2f20ee2eb5b5e8/scripts/camera.lua#L172) and the transitionData is created ([here](https://github.com/veger/TLBE/blob/e34d5ee52662a4eeda0c95366f2f20ee2eb5b5e8/scripts/camera.lua#L171))

This way the tags are not updated when nothing got changed (and the duplicate is fixed)

You could use a new function signature like this:
```
updateChartTags(player, camera, centerPos, zoom, iconSet)
```

and that should be enough to do all calculations and updating I guess.

Reply:
I will take a look at this solution...

Reply:
I went with a variant on this idea. `refreshChartTags(player, camera, iconSet, centerPos, zoom)` takes care of both deleting tags for an icon set and updating their locations.